src/modules/application/application.service.spec.ts

```diff
@@ -1,19 +1,127 @@
+import { Organization } from '../organization/organization.entity';
+import { applicationFixture } from './application.fixture';
 
-// TODO: fix test so pending can be removed
-xdescribe('ApplicationService', () => {
+const chance = new Chance();
+
+const MockLogger = {
+  log: jest.fn(),
+  error: jest.fn(),
+  setContext: jest.fn(),
+  info: jest.fn(),
+  debug: jest.fn(),
+};
+
+const MockOrgService = {
+  getByNamespace: jest.fn(),
+};
+
+describe('ApplicationService', () => {
   let service: ApplicationService;
+  let module: TestingModule;
+  let repo: Repository<Application>;
+  let orgRepo: Repository<Organization>;
+  let organizations: Organization[];
+  let applications: Application[];
+  let queryRunner: QueryRunner;
 
   beforeEach(async () => {
-    const module: TestingModule = await Test.createTestingModule({
-      providers: [ApplicationService],
+    jest.resetAllMocks();
+    module = await Test.createTestingModule({
+      imports: [
+        TypeOrmModule.forRoot(TestDbCOnfig.default as TypeOrmModuleOptions),
+        TypeOrmModule.forFeature([Organization, Application]),
+      ],
+      providers: [
+        ApplicationService,
+        { provide: OrganizationService, useValue: MockOrgService },
+        {
+          provide: Logger,
+          useValue: MockLogger,
+        },
+      ],
     }).compile();
 
+    const dbConnection = module.get(Connection);
+    const manager = module.get(EntityManager);
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    queryRunner = manager.queryRunner = dbConnection.createQueryRunner(
+      'master',
+    );
+    await queryRunner.startTransaction();
+    repo = module.get<Repository<Application>>(getRepositoryToken(Application));
+    orgRepo = module.get<Repository<Organization>>(
+      getRepositoryToken(Organization),
+    );
+
+    organizations = await organizationFixture(orgRepo);
+    applications = await applicationFixture(repo, organizations[0], 2);
+
     service = module.get<ApplicationService>(ApplicationService);
   });
 
-  it('should be defined', () => {
-    expect(service).toBeDefined();
+  afterEach(async () => {
+    await queryRunner.rollbackTransaction();
+    module.close();
+  });
+
+  it('create() it should not save an application with existing namespace and parentOrg ', async () => {
+    const testApp = chance.pickone(applications);
+    const organization = chance.pickone(organizations);
+    const { name, owner, definition, namespace } = testApp;
+
+    MockOrgService.getByNamespace.mockResolvedValueOnce(organization);
+
+    await service.create({
+      name,
+      namespace,
+      owner,
+      definition,
+      parentOrg: organization.namespace,
+    });
+
+    expect(MockLogger.debug).toHaveBeenCalledWith(
+      expect.stringContaining(
+        `Not able to create application: ${namespace} with same parent organization ${organization.namespace} already exists`,
+      ),
+    );
```

Comment:
I think testing the logger is a find approach, but maybe it would be even better to query the db and ensure that there aren't duplicates using the `service`?

Reply:
Not sure that would be useful in this case though, remember we are trying to test if we are able to save or not  and the Logger is only called when we are not able to save it. querying the DB would be an additional call we don't actually need.

Reply:
Hmm, I was suggesting querying the DB in the test to ensure that we don't have a duplicate because that ultimately what we care about. Is that was you're talking about too?
My rationale is that, I know it's not currently doing this, but it would be possible to log that message but still somehow create a duplicate in the DB, in which case the test would be misleading IMO.

Reply:
okay make sense.